### PR TITLE
Ensure all "client" and a few "server" spans are marked as "measured"

### PR DESCRIPTION
**Pull request by tylerbenson**
Rather than doing this for each client integration, I moved the logic to the `ClientDecorator`.  I removed the duplicate asserts from the relevant instrumentation. I also added asserts for it to a bunch of tests to ensure I didn't miss something.

While doing this, I also discovered a few server spans (rmi and grpc) that weren't measured that probably should be. I included those changes here too.

### CODE REVIEW

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/rmi/RmiServerDecorator.java [hunk 1]
```diff
@@ -23,4 +24,10 @@ protected CharSequence spanType() {
   protected CharSequence component() {
     return RMI_SERVER;
   }
+
+  @Override
+  public AgentSpan afterStart(final AgentSpan span) {
+    span.setMeasured(true);
```
Comment (bantonsson) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/rmi/RmiServerDecorator.java:30]: This seems unrelated to the other _client_ changes since it's in `RmiServerDecorator`.

File: dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/server/GrpcServerDecorator.java [hunk 2]
```diff
@@ -59,6 +59,12 @@ protected CharSequence component() {
     return COMPONENT_NAME;
   }
 
+  @Override
+  public AgentSpan afterStart(final AgentSpan span) {
+    span.setMeasured(true);
```
Comment (bantonsson) [inline dd-java-agent/instrumentation/grpc-1.5/src/main/java/datadog/trace/instrumentation/grpc/server/GrpcServerDecorator.java:64]: Same thing here with changes to a server decorator.

**Review by bantonsson — APPROVED** [review]
👍 with only minor questions about why the two server decorators were changed.

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/rmi/RmiServerDecorator.java (hunk 1 above)
Comment (tylerbenson) [reply dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/rmi/RmiServerDecorator.java:30]: Oops, you're right.  I forgot intermingled a couple of server changes here.  Would you prefer I update the title/description or separate to a different PR?

**Review by charliegracie — APPROVED** [review]
